### PR TITLE
Added support for Before and PartOf within unit section of custom systemd configuration

### PR DESCRIPTION
**Pull request by glitchcrab**
### What does this PR do?

- Further extends the fields available for templating systemd units.
- Switch systemd Service ExecStart to an array to support multiple commands (added in #351)

### What is the effect of this change to users?

None, backwards compatible addition which is configured by provider charts.

### How does it look like?

N/A

### Any background context you can provide?

See https://github.com/giantswarm/cluster/pull/351

### Should this change be mentioned in the release notes?

- [x] CHANGELOG.md has been updated (if it exists)

### CODE REVIEW

**Review by Gacko — APPROVED** [review]

File: helm/cluster/templates/clusterapi/_helpers_flatcar.tpl [hunk 1]
```diff
@@ -42,10 +52,18 @@
     {{- end }}
     {{- if .contents.service }}
     [Service]
+    {{- if .contents.service.environment }}
```
Comment (Gacko) [inline helm/cluster/templates/clusterapi/_helpers_flatcar.tpl:55]: You could also repeat this, right? Like, `environment` could be an array?

**Comment by glitchcrab** [comment]
ok I'm done adding stuff, this is ready to be reviewed please :pray:

File: helm/cluster/templates/clusterapi/_helpers_flatcar.tpl (hunk 1 above)
Comment (glitchcrab) [reply helm/cluster/templates/clusterapi/_helpers_flatcar.tpl:55]: yes that's a good point - i'll switch that over

**Comment by glitchcrab** [comment]
I decided that I might as well add support for `EnvironmentFile` while I'm at it too.

**Review by AverageMarcus — APPROVED** [review]

**Review by vxav — APPROVED** [review]

**Review by anvddriesch — APPROVED** [review]

**Pull request merged**